resources/aws_ec2.go

```diff
@@ -1014,6 +994,65 @@ func (obj *AwsEc2Res) prependName() string {
 	return AwsPrefix + obj.GetName()
 }
 
+// waitForStopAndTerminate waits for the instance to stop and waits for it to
+// terminate. If either waiter returns, the instance state is checked, and an
+// awsEc2Event is returned.
+func (obj *AwsEc2Res) waitForStopAndTerminate(ctx context.Context, waitInput *ec2.DescribeInstancesInput) (awsEc2Event, error) {
```

Comment:
do we want to name this the absurd `waitUntilInstanceStoppedOrTerminatedWithContext` to match AWS naming conventions?

Maybe it could also have the same signature? as WaitUntilInstanceStoppedWithContext (so return error)

IOW we pretend it's identical to `WaitUntilInstanceStoppedWithContext` or `WaitUntilInstanceTerminatedWithContext` except it waits for either.